Libraries/LLM/Configuration.swift

```diff
@@ -26,20 +26,21 @@ public enum StringOrNumber: Codable, Equatable, Sendable {
     }
 }
 
-public struct ModelType: RawRepresentable, Codable, Sendable {
-    public let rawValue: String
+private class ModelTypeRegistry: @unchecked Sendable {
 
-    public init(rawValue: String) {
-        self.rawValue = rawValue
-    }
+    // Note: using NSLock as we have very small (just dictionary get/set)
+    // critical sections and expect no contention.  this allows the methods
+    // to remain synchronous.
+    private let lock = NSLock()
 
+    @Sendable
     private static func createLlamaModel(url: URL) throws -> LLMModel {
         let configuration = try JSONDecoder().decode(
             LlamaConfiguration.self, from: Data(contentsOf: url))
         return LlamaModel(configuration)
     }
 
-    private static var creators: [String: (URL) throws -> LLMModel] = [
+    private var creators: [String: @Sendable (URL) throws -> LLMModel] = [
```

Comment:
This is the key -- we want to mark it as storing thread safe (Sendable) closures.